src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -34,6 +35,10 @@ class LazyList[+A](expr: => LazyList.Evaluated[A])
         case Some((hd, tl)) => s"$hd #:: $tl"
       }
     else "LazyList(?)"
+
+  // HACK Needed because we inherit from two implementations (one from LinearSeq and one from SeqLikeFromIterable)
+  override def drop(n: Int): LazyList[A] = super.drop(n)
```

Comment:
I just added a commit that solves the issue. The solution is to *not* automatically implement `drop` in terms of `head` and `tail` but provide this implementation as a `LinearSeqLikeOpt` trait that can optionally be mixed into a concrete collection implementation.